sklearn/manifold/tests/test_isomap.py

```diff
@@ -15,65 +10,75 @@
 from sklearn import preprocessing
 from sklearn.datasets import make_blobs
 from sklearn.metrics.pairwise import pairwise_distances
-from sklearn.utils._testing import assert_allclose, assert_allclose_dense_sparse
-
+from sklearn.utils._testing import (
+    assert_almost_equal,
+    assert_allclose,
+    assert_allclose_dense_sparse,
+    assert_array_equal,
+)
 from scipy.sparse import rand as sparse_rand
 
 eigen_solvers = ["auto", "dense", "arpack"]
 path_methods = ["auto", "FW", "D"]
 
+# TODO: Isomap fit/fit_transform must preserve dtype
+
```

Comment:
Not sure we want to use global_dtype when we know that the transformer converts anyway. It means that we are just testing twice the same thing. Maybe it's better to do that as part of the PR that enable preserve_dtypes for isomap ?

Reply:
Definitely. I was not thinking of addressing it in this PR. I missed https://github.com/scikit-learn/scikit-learn/issues/11000 and I think this comment can be removed.

Reply:
I meant that I'm not sure enabling global_dtypes is worth for Isomap since it converts to float64 anyway, so we're not really testing on float32. I think it should be enabled at the same time as isomap stops converting float32

Reply:
If #24714 is merged, then I think we can remove this comment since fit/fit_transform dtype preservation will already be tested.
```suggestion
```

What do you think? I just have updated this dangling PR in this regard.